src/js.c

```diff
@@ -476,6 +476,15 @@ _setHeadersFromOptions(natsMsg *msg, jsPubOptions *opts)
             s = natsMsgHeader_Set(msg, jsExpectedLastSubjSeqHdr, temp);
         }
     }
+    if ((s == NATS_OK) && (opts->MsgTTL != 0))
+    {
+        if (opts->MsgTTL < 0)
+            s = nats_setError(NATS_INVALID_ARG, "option 'MsgTTL' (%" PRId64 ") cannot be negative", opts->MsgTTL);
+        if (s == NATS_OK)
```

Comment:
Should be a `else` here instead.

Reply:
Thx!!! Cut-paste error, was returning there before.